### DISCUSSION TITLE
Can't build a SvelteKit app if Pocketbase is not running (ECONNREFUSED)

### DISCUSSION THREAD

**Discussion by nikolaydyankov** [pocketbase/pocketbase#2483] [Q&A] [answered]
I can't build my SvelteKit app if pocketbase is not running. I'm not sure if this is normal, I don't think it should be the case. This is the error that I'm getting:

```bash
ClientResponseError 0: Something went wrong while processing your request.
    at new ClientResponseError (file:///website/node_modules/pocketbase/dist/pocketbase.es.mjs:1:2306)
    at file:///website/node_modules/pocketbase/dist/pocketbase.es.mjs:1:45357
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///website/.svelte-kit/output/server/chunks/pocketbase.js:4:1 {
  url: '',
  status: 0,
  response: {},
  isAbort: false,
  originalError: TypeError: fetch failed
      at fetch (website\node_modules\undici\index.js:109:13)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async file:///website/.svelte-kit/output/server/chunks/pocketbase.js:4:1 {
    cause: Error: connect ECONNREFUSED 127.0.0.1:8090
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
      errno: -4078,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '127.0.0.1',
      port: 8090
    }
  }
}

Node.js v18.14.2
[vite-plugin-sveltekit-compile] Failed with code 1
✓ built in 1.60s
error during build:
Error: Failed with code 1
    at ChildProcess.<anonymous> (file:///website/node_modules/@sveltejs/kit/src/utils/fork.js:68:13)
    at ChildProcess.emit (node:events:513:28)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12)
```

Starting Pocketbase and then doing `npm run build` in another terminal doesn't throw this error and it builds successfully.
I tried changing "127.0.0.1" with "localhost". Also I'm using adapter-node if it makes any difference. What am I doing wrong?

**Comment by ganigeorgiev** [comment, marked as answer]
It seems that the build process is trying to make request to the PocketBase server. It may be related to https://github.com/sveltejs/kit/issues/933.

Do you have [prerendering](https://kit.svelte.dev/docs/glossary#prerendering) enabled?

**Reply by nikolaydyankov** [reply]
I have `export const prerender = false` both in my `layout.ts` and `layout.server.ts`, also in svelte.config.js:

```javascript
const config = {
	preprocess: vitePreprocess(),
	kit: {
		adapter: adapter(),
		prerender: {
			crawl: false,
			entries: []
		}
	}
}
```

**Reply by ganigeorgiev** [reply]
Could you provide a minimal reproducible repo?

I don't think the error is caused by the JS SDK and probably will fail with any other inaccessible fetch resource.

**Reply by nikolaydyankov** [reply]
I don't know how legal is this, but it works:

```javascript
// $lib/pocketbase.ts
import { building } from '$app/environment'

if (!building) {
    // initialize Pocketbase here...
}
```

**Reply by nikolaydyankov** [reply]
> Could you provide a minimal reproducible repo?
> 
> I don't think the error is caused by the JS SDK and probably will fail with any other inaccessible fetch resource.

Yes I think this has nothing to do with the JS SDK. Thank you for taking your time to reply anyway, you're the best.

**Reply by ganigeorgiev** [reply]
> I don't know how legal is this, but it works:

I think it is OK since it seems also to be recommended in the [docs](https://kit.svelte.dev/docs/building-your-app#during-the-build):

> SvelteKit will load your +page/layout(.server).js files (and all files they import) for analysis during the build. **Any code that should not be executed at this stage must check that building from [$app/environment](https://kit.svelte.dev/docs/modules#$app-environment) is false**

I remember testing it and didn't stumble on this issue, although it was before the SvelteKit v1 release and the `kit.prerender` option was only true/false, so things may have change since then or I didn't tested it properly. I'll check it again sometime later this week and eventually will add a note in JS SDK SvelteKit example.
